templates/inspec/nested_object.erb

```diff
@@ -40,23 +40,28 @@ module GoogleInSpec
 <% end # property.nested_properties.each -%>
 
 <% end # if property.nested_properties? -%>
-        def initialize(args = nil)
+        def initialize(args = nil, parent_identifier = nil)
```

Comment:
It seems like parent_identifier is always nil

Reply:
The parent_identifier is passed in when the top level resource creates nested objects, passing them the result of the resource's `to_s` method. That will be something like `Autoscaler #{autoscaler_name}` and then nested objects will append their object name to that identifier.

See changes in `provider/inspec.rb` for where this is initially passed in.